src/Form/SettingsForm.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\ezcontent\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Class SettingsForm.
+ *
+ * @package Drupal\ezcontent\Form
+ */
+class SettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'ezcontent.settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('ezcontent.settings');
+
+    $form['revision'] = [
+      '#type'  => 'fieldset',
+      '#title' => 'Hide Entity Revision Field',
+      '#open'  => TRUE,
+    ];
+
+    $form['revision']['media'] = [
+      '#type'  => 'fieldset',
+      '#title' => 'Media',
+      '#description' => $this->t('Please select media type for which revision information will be disabled.'),
+      '#open'  => TRUE,
+    ];
+
+    $media_types = \Drupal::entityTypeManager()->getStorage('media_type')->loadMultiple();
+    foreach ($media_types as $media_type_name => $value) {
+      $form['revision']['media'][$media_type_name] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t($media_type_name),
+        '#default_value' => $config->get($media_type_name),
+      ];
+
+    }
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
```

Comment:
Do we need this method to be overwritten ?

Reply:
Yes sir in order to keep the values on form